src/c_kzg_4844.c

```diff
@@ -1766,39 +1799,76 @@ C_KZG_RET LOAD_TRUSTED_SETUP(
  * @param[out] out Pointer to the loaded trusted setup data
  * @param[in]  in  File handle for input
  */
-C_KZG_RET LOAD_TRUSTED_SETUP_FILE(KZGSettings *out, FILE *in) {
+C_KZG_RET load_trusted_setup_file(KZGSettings *out, FILE *in) {
+    C_KZG_RET ret;
     int num_matches;
-    uint64_t i;
-    uint8_t g1_bytes[TRUSTED_SETUP_NUM_G1_POINTS * BYTES_PER_G1];
-    uint8_t g2_bytes[TRUSTED_SETUP_NUM_G2_POINTS * BYTES_PER_G2];
+    size_t size;
+    uint8_t *g1_bytes = NULL;
+    uint8_t *g2_bytes = NULL;
 
     /* Read the number of g1 points */
-    num_matches = fscanf(in, "%" SCNu64, &i);
-    CHECK(num_matches == 1);
-    CHECK(i == TRUSTED_SETUP_NUM_G1_POINTS);
+    uint64_t num_g1_points;
+    num_matches = fscanf(in, "%" SCNu64, &num_g1_points);
+    if (num_matches != 1) {
+        ret = C_KZG_BADARGS;
+        goto out;
+    }
```

Comment:
The code currently checks that n1 matches either the mainnet or minimal setting.
While that's an OK thing to do, the code actually should work in a more general setting and the current behaviour is surprising; my expectation from reading the functions' doc would be that it worked with any value of n1, n2.

As it stands, the fact that it (currently?) only works for n1, n2 matching these settings needs to be added to the documentation.

Reply:
Good point. Would you prefer we update it to be more generic or add a note to the doc?

Reply:
I would prefer adding a note.

Reply:
Agreed that this should be a doc fix, instead of a "make it more generic" fix.